### PR TITLE
Editor view UI

### PR DESCRIPTION
**Pull request by k88hudson**
This is just a UI pass, as well as some components:

- Button
- ColorGroup
- Range

there are some performance issues with range around model updating

### CODE REVIEW

File: www_src/components/color-group/color-group.jsx [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+  },
+  getInitialState: function () {
+    return {
+      value: 'transparent'
+    };
+  },
+  render: function () {
+    var linkState = this.props.linkState || this.linkState;
+    var valueLink = linkState(this.props.id);
+    function onChange(e) {
```
Comment (gvn) [inline www_src/components/color-group/color-group.jsx]: Seems weird to have this be in the render function. I'd make it a method instead...

File: www_src/components/range/range.jsx [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  },
+  getInitialState: function () {
+    return {
+      value: 100
+    };
+  },
+  render: function () {
+    var linkState = this.props.linkState || this.linkState;
+    var valueLink = linkState(this.props.id);
+    return (<div className="range">
```
Comment (gvn) [inline www_src/components/range/range.jsx]: Kind of a nit, but I find render returns to be more readable like this:

```javascript
return (
  <div className="range">
    <input min={this.props.min} max={this.props.max} type="range" valueLink={valueLink}/>
    <div className={'range-summary' + (parseInt(valueLink.value, 0) === this.props.min ? ' min' : '')}>{valueLink.value}{this.props.unit}</div>
  </div>
);
```

File: www_src/components/range/range.less [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+  }
+
+  input[type=range]::-webkit-slider-thumb {
+    -webkit-appearance: none;
+    border: none;
+    height: 30px;
+    width: 30px;
+    border-radius: 36px;
+    background: #FFF;
+    cursor: pointer;
```
Comment (gvn) [inline www_src/components/range/range.less]: Is this for future desktop use?

File: www_src/static/img/change-image.svg [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+                </g>
+            </g>
+        </g>
+    </g>
+</svg>
```
Comment (gvn) [inline www_src/static/img/change-image.svg]: Recommend running this through `svgo` to optimize.

**Comment by gvn** [comment]
Some nits/suggestions, but R+ :fire:

File: www_src/components/range/range.less (hunk 3 above)
Comment (k88hudson) [inline www_src/components/range/range.less]: yeah idk, i figured why not

**Comment by gvn** [comment]
Oh, but FWIW, I wasn't able to get this running on device. I saw a lot of "Webpage not available" messages.

**Pull request merged**